src/mem/cache/replacement_policies/sim_object_unit_tester.cc

```diff
@@ -0,0 +1,193 @@
+#include "mem/cache/replacement_policies/sim_object_unit_tester.hh"
+
+#include <iostream>
+
+#include "base/trace.hh"
+#include "mem/cache/replacement_policies/fifo_rp.hh"
+#include "mem/cache/replacement_policies/replaceable_entry.hh"
+
+namespace gem5
+{
+namespace replacement_policy
+{
+
+SimObjectUnitTester::SimObjectUnitTester(
+  const SimObjectUnitTesterParams& params):
+  SimObject(params),
+  nextEvent([this](){ processNextEvent(); }, name() + "nextEvent"),
+  replacementPolicy(params.replacement_policy),
+  numEntries(params.num_entries),
+  replacementPolicyName(params.name)
+{}
+
+void
+SimObjectUnitTester::processNextEvent()
+{
+  std::cout << "tick: " << curTick() <<
+  ", Hello from SimObjectUnitTester::processNextEvent!" << std::endl;
+
+  // create a new entry and put it in candidates
+  ReplaceableEntry* temp = new ReplaceableEntry();
+  temp->replacementData = replacementPolicy->instantiateEntry();
+  replacementPolicy->reset(temp->replacementData); // record insertion tick
+  // std::cout<< "Inserted tick: " <<
+  //std::static_pointer_cast<FIFO::FIFOReplData>(temp->replacementData)
+  //->tickInserted <<std::endl;
+  candidates.push_back(temp);
+
+  if (numEntries > 0){
+    schedule(nextEvent, curTick() + 500);
+    numEntries--;
+  }
+  else {
+    bool correct = checkCorrectness(replacementPolicy);
+    // free memory
+    freeCandidates();
+    if (correct){
+      exit(0);
+    }
+    exit(1);
```

Comment:
Id remove this. I think the best thing to do here is just to have the SimObject print something is to throw a `warn` if there's unexpected behavior(e.g. "Replacement policy 'FOFI''s victim was not first in!'). Then, in our testlib, we just check the output and report an report a test failure if there's anything printed we weren't expected.